### PR TITLE
TEP-0090: Resolve `PipelineTask` with `Custom Task`

### PR DESCRIPTION
**Pull request by jerop**
# Changes

[TEP-0090: Matrix][tep-0090] proposed executing a `PipelineTask` in parallel `TaskRuns` and `Runs` with substitutions from combinations of `Parameters` in a `Matrix`.

This change resolves a `PipelineTask` with a matrixed `Custom Task`.

[tep-0090]: https://github.com/tektoncd/community/blob/main/teps/0090-matrix.md

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Release notes block below has been filled in (if there are no user facing changes, use release note "NONE")

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by jerop** [comment]
@abayer @dibyom @lbernick please take a look

**Comment by abayer** [comment]
Unit test failure seems legit.

**Comment by jerop** [comment]
@abayer ah was a last minute change before pushing, fixed it 👍🏾

**Comment by jerop** [comment]
`level=error msg="Timeout exceeded: try increasing it by passing --timeout option"` 

/test pull-tekton-pipeline-build-tests

File: pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go [hunk 1]
```diff
@@ -525,6 +526,15 @@ func ResolvePipelineTask(
 	}
 	rpt.CustomTask = isCustomTask(ctx, rpt)
 	switch {
+	case rpt.IsCustomTask() && rpt.IsMatrixed():
```
[the remark this answers is not in the document]
Comment (jerop) [reply pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go:529]: this is logic for creating Runs from matrixed PipelineTasks with Custom Tasks 

we already implemented creating TaskRuns from matrixed PipelineTasks with Tasks - https://github.com/tektoncd/pipeline/pull/4990/files#diff-39d9baa708e379638f742c8b9fd36d9d15c64c3386de958b5a3f309ca8fa037a

File: pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go [hunk 2]
```diff
@@ -3201,6 +3201,106 @@ func TestResolvePipelineRunTask_WithMatrix(t *testing.T) {
...
+	getTask := func(ctx context.Context, name string) (v1beta1.TaskObject, error) { return task, nil }
+	getTaskRun := func(name string) (*v1beta1.TaskRun, error) { return &trs[0], nil }
+	getRun := func(name string) (*v1alpha1.Run, error) { return runsMap[name], nil }
+
+	for _, tc := range []struct {
+		name string
+		pt   v1beta1.PipelineTask
+		want *ResolvedPipelineTask
+	}{{
+		name: "custom task with matrix - single parameter",
```
[the remark this answers is not in the document]
Comment (lbernick) [reply pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go:3265]: Actually, never mind, these aren't treated as part of the same pipelinerun

**Review by lbernick — APPROVED** [review]

**Comment by lbernick** [comment]
/lgtm

since andrew already approved

**Pull request merged**